### ISSUE TITLE
RFM69CW on this PCB

### ISSUE THREAD

**Issue by ropolik** [uChip/Moteino_Peripherals_Half-Shield#1]
Hi uChip,

this board is excellent! Do you have design of board for RFM69CW ?
I have RFM69CW and want use RFM69CW on this board and like me this half-shiled with serial flash which I have too.

Many thanks for answer

**Comment by uChip** [comment]
I have not worked with the RFM69CW, however from a very brief scan of the datasheet it appears to be very similar from a signal I/O perspective.  I would be willing to create a RFM69CW package and board layout, but there would be caveats.  The first would be that you would need to test and verify the board yourself since I don't have an RFM69CW, nor do I really have time for another project just now.  The other caveat would be that I have no idea if the RFM69CW programs the same as the other modules.  Any software modifications required, you would need to do yourself.  If that is acceptable, let me know and I'll take a crack at the package and board.
- Chip

**Comment by uChip** [comment]
Oops.  Leaving issue open until ropolik responds.

**Comment by ropolik** [comment]
RFM69CW by function is same like RFM69W. But letter C indicates that RFM69CW is backward Compatibility with pinout of RFM12B.
I can try it (I have several RFM69CW ), but I do not have the PCB or design of PCB in Eagle.
I can send the library of RFM69CW for Eagle.

**Comment by uChip** [comment]
Is it footprint compatible with the 12B?  I have an Eagle library for that.  Or go ahead and send what you have.  Thanks.

**Comment by ropolik** [comment]
Yes. Footprint is compatible with the RFM12B. Where I can send the library to you?

Right side on scheme from bottom to top: Pin 1-ANT, 2-VCC, 3-GND, 4-DIO3, 5-MOSI, 6-SCK, 7-NSS
Left side on scheme from bottom to top: Pin 14-GND, 13-RESET, 12-DIO5, 11-DIO1, 10-DIO2, 9-DIO0, 8-MISO.
RFM69CW and RFM12B is have same dimensions of case and same footprint.

http://www.hoperf.com/rf/fsk_module/RFM69CW.htm

Thanks for your interest and assistance!

**Comment by ropolik** [comment]
http://www.hoperf.com/rf/fsk_module/RFM69CW.htm

**Comment by uChip** [comment]
Ok.  You will soon find a new schematic and board files in this repo.  Check both carefully.  I used the RFM12B footprint from the orginal Low Power Lab breakout board as I already had that lying around.  If the RFM69C is truly compatible then all should be good.  Again, check the files carefully as I have not built the board.  You are welcome to take the board file and submit it to fab, or I can submit it and share it out.  Let me know if you want me to do that.
I'll leave this open for a while to let you confirm the design.

**Comment by ropolik** [comment]
Yes. If I found on this project new design of the PCB then I am try made in fab the new PCB with RFM69CW.  I try the new PCB and write feedback of results.

Many thanks for new early PCB :-)

Best Regards

**Comment by ropolik** [comment]
I am download new design of PCB. By first look I see than ground of antenne is not connected to ground on PCB.
I do not know which layer of PCB solves grounding. I do not know how is connected to ground next points.
Missing wire on PCB between GND of header (pin 2) and C4, R6 on Top side of PCB.
On the bottom side of PCB is missing wire between GND of header (pin 2) and pin 1 of U2 (LDO regulator) - via hole of pin 1 of U2.
Not connected ground wire of C2 and C3 on top side of PCB. May be connected to GND on bottom side of PCB. And next grounded pins, pads, etc.

I have some LDO AMS1117-3.3V in case SOT-223 which is have reversed IN and OUT VCC. From left to right from view of top case pinout is 1-GND, 2-OUT, 3-IN. Would you make PCB design with this LDO?

Many, many thanks for your answer and assistance.

**Comment by uChip** [comment]
Grounds are connected by a flood.  Click on the "Ratsnest" button to have the flood redrawn.  Also, I just checked the 1117 and it has the same pinout as the 1701, but they are not the same package.  The 1117 is in the SparkFun library.  You are welcome to change it if you like.

**Comment by ropolik** [comment]
Yes. I want use AMS1117 LDO. Would you make another design of PCB with AMS1117 3.3v 
?

**Comment by ropolik** [comment]
I am look to the sheet again (http://www.advanced-monolithic.com/pdf/ds1117.pdf). AMS1117 in package SOT-223 from top view is have on middle pin V-OUT (output) and on the right pin is have V-IN (input).
MCP 1701 (sheet http://ww1.microchip.com/downloads/en/DeviceDoc/21874b.pdf) from top view is have on middle pin V-IN (INPUT) and on the right pin is have V-OUT (output).

You say truth that package SOT-223 (AMS1117) is bigger than package SOT-89 (MCP 1701). I am sorry, my error.

Would you make another design of PCB with AMS1117 3.3v (package SOT-223 and midle and right pin is different than MCP 1701) ?

Many, many thanks for your assistance!

Best Regards

**Comment by uChip** [comment]
I have made the change but the layout is now quite cramped.  I don't like it.  You should take the files with the change then I will take them down again.
- Chip

**Comment by ropolik** [comment]
Yes, I have the new files with PCB and scheme. Will do you another new variant of PCB?

**Comment by uChip** [comment]
The latest version posted yesterday has the change to the 1117 voltage regulator.  I do not plan on another version.
- Chip

**Comment by ropolik** [comment]
Ok. I asked you in reaction that you wrote: " I don't like it. ".

I try made the new PCB and test it. I write later how this new PCB is works.

Many thanks for your help!

**Comment by uChip** [comment]
Ah.  I understand now.
Improving the layout could probably achieved with more work, but as I noted earlier, I don't really have time for a more involved project right now.  Nor do I see a great advantage to using the RFM69C and 1117 over those of my original layout.  I hope that the board works for you but I will pull down this variant now that you have it.  I don't want to confuse other readers with variants that are not really supported.  Good luck with your project.
- Chip

**Comment by ropolik** [comment]
uChip, I printed the PCB and I placed the components (RFM69CW and LDO AMS1117). I would need more space on the PCB between RFM and antenna input and next space between RFM and AMS1117. On the photo is see the place which I need more space. Would you made PCB with this modifications?
![rfm69cw-moteino](https://cloud.githubusercontent.com/assets/12201545/10291979/e54657aa-6bac-11e5-9c06-1ca925733d39.jpg)

Do you have PayPal account? I want send to you some money for your help.

**Comment by uChip** [comment]
Yes, that is the cramping I spoke of.  Although it could be assembled as is, I think the best solution is to make the board about 2 tenths of an inch taller.  I will take a look at doing that.

Since I am not verifying the design or affirming that it will even work, it would be inappropriate to take money for it.  Thank you for the thought.
- Chip

**Comment by uChip** [comment]
I extended the board by 0.1 inches and moved the radio and antenna connections up to spread things apart.  Print this one out as you did before and check the size.  I hope it is enough but if not I will move it some more.
Instead of adding the board to the repo again, I have attached it here.  To attach the file I had to change the file type to text.  After you download it change the file type back to brd so that Eagle recognizes it.
[MPHS-RFM69C.txt](https://github.com/uChip/Moteino_Peripherals_Half-Shield/files/9038/MPHS-RFM69C.txt)
- Chip

**Comment by uChip** [comment]
I haven't heard back from ropolik. I will open this issue again in case that is needed to get messages sent.  ropolik, please see previous comment and respond if new layout works.
- Chip
